### PR TITLE
Added basic logic to button states + API integration

### PR DESCRIPTION
**Pull request by gunbux**
## Changes Made
- Added logic for the button states based on the status of the gym
- Attached API calls to frontend

### CODE REVIEW

File: frontend/src/components/Button.tsx [hunk 1]
```diff
@@ -34,11 +34,12 @@ type Props = {
   state: 'primary' | 'secondary'
   text: string
   onClick: React.MouseEventHandler<HTMLButtonElement>
+  disabled?: boolean
 }
 
 function ButtonComponent(prop: Props) {
   return (
-    <Border onClick={prop.onClick} state={prop.state}>
+    <Border onClick={prop.onClick} state={prop.state} disabled={prop.disabled || false}>
```
Comment (cjunxiang) [inline frontend/src/components/Button.tsx:42]: `prop.disabled || false` can be represented just by `prop.disabled`

File: frontend/src/routes/GymPage/index.tsx [hunk 2]
```diff
@@ -49,22 +57,77 @@ export default function GymPage({ onLeftClick }: { onLeftClick?: () => void }) {
...
+    dispatch(getGymStatus())
+    dispatch(getUserDetail())
+    // dispatch(getGymHistory())
+  }, [])
 
   const [currentTab, setCurrentTab] = useState<number>(1)
   const sections = ['Gym', 'History']
   const history = useHistory()
+  //TODO: export this
+  const buttonText = ['Key With Me', 'Return Key', 'Close Gym', 'Open Gym']
```
Comment (cjunxiang) [inline frontend/src/routes/GymPage/index.tsx:70]: maybe can represent in a map, so to access you don't need index numbers to get them, which is another area for errors.

actionButtonLabel.keyWithMe: 'Key With Me'
actionButtonLabel.returnKey: 'Return Key'

File: frontend/src/routes/GymPage/index.tsx [hunk 3]
```diff
@@ -49,22 +57,77 @@ export default function GymPage({ onLeftClick }: { onLeftClick?: () => void }) {
...
+
+  function getButtonStates(): ButtonStates {
+    if (!gymStatus.gymIsOpen) {
+      if (gymStatus.keyHolder.displayName == name) {
+        return { keyWithMe: false, returnKey: true, toggleGym: true }
```
Comment (cjunxiang) [inline frontend/src/routes/GymPage/index.tsx:75]: consider abstracting them out into a variable for easy reading.

const gymOpenBySomeoneElseButtonStates = {..false ..true ..true}
const gymOpenByMeButtonStates = {......}
const gymClosedButtonStates = {...}

if (xxx) {
  return gymClosedButtonStates

**Comment by cjunxiang** [comment]
Overall good development & scope. But small things can be improved -- such as what data structure to use -- map? array? off hand, you can think about 
- writing:  is writing to this data structure easy? Do I require additional variables to be able to write to it properly? Do i need it to be sorted when i write into it?
- accessing: is reading this data structure prone to error? e.g. array[idx], you need to have a few more "blind" variables, such as idx. 
- editting data: will i be able to find smth easily? <- and thus mutate it easily?

**Review by zzhikai — APPROVED** [review]
Great work! LGTM. Though we might have to reverse the order in which the gymHistory appears to latest from the top.

Screenshot 2022-02-22 at 4 10 40 PM

**Pull request merged**